### PR TITLE
chore: add deployment based on tag

### PR DESCRIPTION
**Pull request by warengonzaga**
Add the vercel deployment action based on tags.

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Production Tag Deployment
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+on:
+  push:
+    # Pattern matched against refs/tags
+    tags:        
+      - '*' # Push events to every tag not containing /
```
Comment (jofftiquez) [inline .github/workflows/deployment.yml:9]: Kindly change to `v*` to match semantic versioning pattern

**Review by jofftiquez — CHANGES_REQUESTED** [review]
Just one small change

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (warengonzaga) [reply .github/workflows/deployment.yml]: done, thanks @jofftiquez

**Review by jofftiquez — APPROVED** [review]
Looks good, thanks 🙏

**Pull request merged**